### PR TITLE
bpo-40077: Convert _queuemodule to use heap types

### PR DESCRIPTION
**Pull request by erlend-aasland**
@vstinner / @corona10, would you mind reviewing this?

https://bugs.python.org/issue40077

Automerge-Triggered-By: GH:pitrou

### CODE REVIEW

**Comment by erlend-aasland** [comment]
FYI, rebased onto master, applied `_PyType_GetModuleByDef` in order to migrate from global to module state.

File: Modules/_queuemodule.c [hunk 1]
```diff
@@ -155,6 +169,9 @@ simplequeue_pop_item(simplequeueobject *self)
 
 /*[clinic input]
 _queue.SimpleQueue.get
+
+    cls: defining_class
```
Comment (pitrou) [inline Modules/_queuemodule.c:173]: Hmm... will this appear in docstrings and generated docs?

File: Modules/_queuemodule.c [hunk 2]
```diff
@@ -358,42 +359,51 @@ PyDoc_STRVAR(queue_module_doc,
...
-    -1,
-    NULL,
-    NULL,
-    NULL,
-    NULL,
-    NULL
+    .m_base = PyModuleDef_HEAD_INIT,
+    .m_name = "_queue",
+    .m_doc = queue_module_doc,
+    .m_size = sizeof(simplequeue_state),
```
Comment (pitrou) [inline Modules/_queuemodule.c:388]: Shouldn't you define `m_clear` and `m_free` to clear the `EmptyError` and `SimpleQueueType` members of the state?

File: Modules/_queuemodule.c (hunk 1 above)
Comment (erlend-aasland) [reply Modules/_queuemodule.c:173]: No

File: Modules/_queuemodule.c (hunk 2 above)
Comment (erlend-aasland) [reply Modules/_queuemodule.c:388]: Oops, of course! Comin' up.

**Comment by erlend-aasland** [comment]
I have made the requested changes; please review again.

**Review by pitrou — APPROVED** [review]
+1, thank you @erlend-aasland

**Comment by erlend-aasland** [comment]
Thanks for reviewing, @pitrou !

**Comment by vstinner** [comment]
@pitrou: You merged the commit message with the sentence:

> @vstinner / @corona10, would you mind reviewing this?

And I'm now getting emails from cpython forks, like this commit:

https://github.com/JelleZijlstra/cpython/commit/01c6aa43dc56b3b64d584c58a49c86f816c05a91

**Comment by pitrou** [comment]
Ah, sorry. I get this sometimes when my name is mentioned too :-)

**Comment by vstinner** [comment]
> Ah, sorry. I get this sometimes when my name is mentioned too :-)

Well, the "automerge" label causes a similar feature and it's part of our workflow. I'm not sure how to ignore such emails. @pganssle has such issue: https://twitter.com/pganssle/status/1379161385863643155

**Comment by erlend-aasland** [comment]
It's partly my fault; I cc'd you in the PR text instead of in a separate comment/post. After this incident, I've made sure I don't mention anyone in the PR text. Maybe it could be mentioned in the Dev Guide? Maybe we could make the CI prevent it by rewriting the commit message (if needed) upon merging.

**Comment by vstinner** [comment]
I asked how to not get spammed on python-committers: https://mail.python.org/archives/list/python-committers@python.org/thread/OCVQ6IYOVTVYW7ADDNOHKR7JDRBYLM3T/

**Pull request merged**